### PR TITLE
Move debug processor to exporters, because no next.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Actually the debug processor was moved but github does not see it as a moved file.

**Comment by bogdandrutu** [comment]
To add metrics I am waiting for https://github.com/census-instrumentation/opencensus-service/pull/442 to be merged.

File: exporter/debugexporter/debug_processor.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2019, OpenCensus Authors
```
Comment (sjkaris) [inline exporter/debugexporter/debug_processor.go]: nit: maybe rename this file debug_processor.go -> exporter.go or debug_expoter.go ?

File: cmd/occollector/app/collector/processors.go [hunk 2]
```diff
@@ -247,7 +248,7 @@ func startProcessor(v *viper.Viper, logger *zap.Logger) (processor.SpanProcessor
 	_ = metricsExporters
 
 	if builder.DebugProcessorEnabled(v) {
-		dbgProc := processor.NewNoopSpanProcessor(logger)
+		dbgProc := processor.NewTraceExporterProcessor(debugexporter.NewDebugTraceDataExporter(logger))
```
Comment (pjanotti) [inline cmd/occollector/app/collector/processors.go]: nit: naming: debugexporter.NewTraceDataExporter

File: exporter/debugexporter/debug_processor.go (hunk 1 above)
Comment (pjanotti) [reply exporter/debugexporter/debug_processor.go]: +1

File: exporter/debugexporter/debug_processor.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+// NewDebugTraceDataExporter creates an exporter.TraceDataExporter that just drops the
+// received data and logs debugging messages.
+func NewDebugTraceDataExporter(logger *zap.Logger) exporter.TraceDataExporter {
+	return &debugExporter{logger: logger}
+}
+
+// NewDebugMetricsDataExporter creates an exporter.MetricsDataExporter that just drops the
+// received data and logs debugging messages.
+func NewDebugMetricsDataExporter(logger *zap.Logger) exporter.MetricsDataExporter {
```
Comment (pjanotti) [inline exporter/debugexporter/debug_processor.go]: nit: naming: NewMetricsDataExporter

File: internal/collector/processor/processor.go [hunk 4]
```diff
@@ -34,29 +31,6 @@ type SpanProcessor interface {
...
 }
 
-// An initial processor that does not sends the data to any destination but helps debugging.
-type debugSpanProcessor struct{ logger *zap.Logger }
-
-var _ SpanProcessor = (*debugSpanProcessor)(nil)
-
-func (sp *debugSpanProcessor) ProcessSpans(td data.TraceData, spanFormat string) error {
-	if td.Node == nil {
-		sp.logger.Warn("Received batch with nil Node", zap.String("format", spanFormat))
```
Comment (pjanotti) [inline internal/collector/processor/processor.go]: I kind of miss that enabling the debug exporter we were able to log the format of the spans reaching the service, I guess we will go with metrics to identify that now...

File: cmd/occollector/app/collector/processors.go (hunk 2 above)
Comment (pjanotti) [reply cmd/occollector/app/collector/processors.go]: starting to think that we could short `TraceDataExporter` to `TraceExporter` (symmetrically for metrics)

Comment (bogdandrutu) [reply cmd/occollector/app/collector/processors.go]: done.

File: exporter/debugexporter/debug_processor.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/debugexporter/debug_processor.go]: done.

File: exporter/debugexporter/debug_processor.go (hunk 3 above)
Comment (bogdandrutu) [reply exporter/debugexporter/debug_processor.go]: done.

File: internal/collector/processor/processor.go (hunk 4 above)
Comment (bogdandrutu) [reply internal/collector/processor/processor.go]: +1 metrics will be good.

File: cmd/occollector/app/collector/processors.go (hunk 2 above)
Comment (bogdandrutu) [reply cmd/occollector/app/collector/processors.go]: Done second comment.

**Pull request merged**